### PR TITLE
Critical fix of the table assembly

### PR DESCRIPTION
**Pull request by Louis-Boulet-HC**
- Fix to properly iterate through rows and cut into columns/cells
- Pixel threshold for filtering out rows & columns has been increased to 4
   - From 2 beforehand, was still getting some extra thin columns (and rows) filtering through
   - Continue monitoring and adjust accordingly
- Cleaning of the get_cropped_rows() & get_cropped_columns() functions
- Addition of filters to remove extra rows & columns at the start and end of the outputted dataframe
- Fixed handling of column row (no more duplication of header as first row)
- Added condition to add empty row if table is only one row (only a header row)
    - to reflect new column header row changes

NOTE: Dimensions are (seemingly) now being completely properly read and outputted  :)

### CODE REVIEW

File: tests/table_test.py [hunk 1]
```diff
@@ -28,3 +28,47 @@ def test_to_excel():
     os.remove("all_excel.xlsx")
     assert not os.path.exists("all_excel.xlsx")
 
+def test_calculate_row_column_limits():
+    # TODO: Make a test for calculate_row_column_limits
+    assert False
```
Comment (pgaviganHC) [inline tests/table_test.py:33]: Test case?

File: tests/table_test.py [hunk 2]
```diff
@@ -28,3 +28,47 @@ def test_to_excel():
...
+
+def test_get_cropped_rows():
+    # TODO: Make a test case for get_cropped_rows
+    assert False
```
Comment (pgaviganHC) [inline tests/table_test.py:37]: Test case?

File: tests/table_test.py [hunk 3]
```diff
@@ -28,3 +28,47 @@ def test_to_excel():
...
+
+def test_get_cropped_columns():
+    # TODO: Make a test case for get_cropped_columns
+    assert False
```
Comment (pgaviganHC) [inline tests/table_test.py:41]: Test case?

File: tests/table_test.py [hunk 4]
```diff
@@ -28,3 +28,47 @@ def test_to_excel():
...
+    from table_processing.Table import Table
+
+    file_path = "./tests/resources/simple_table.PNG"
+    table_image = Image.open(file_path).convert("RGB")
+    width, height = table_image.size
+    table_image.resize((int(width*0.5), int(height*0.5)))
+    table = Table(image = table_image)
+
+    # TODO: Correct this test case to check the table contents
+    assert False
```
Comment (pgaviganHC) [inline tests/table_test.py:55]: Please update this test to verify the contents of the table.

**Review by pgaviganHC — COMMENTED** [review]
Great progress!

I'm still keen on seeing the completed test cases.

**Review by fnwaohaHC — APPROVED** [review]

**Comment by Louis-Boulet-HC** [comment]
> Please advise what to run so that I can replicate your results and verify that things are working as expected on my system.

just the benchmark_pipeline.py

**Review by zgaffarHC — APPROVED** [review]

**Pull request merged**